File: app/src/main/java/in/smslite/utils/ThreadUtils.java
package in.smslite.utils;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.os.AsyncTask;
import android.provider.Telephony;
import android.util.Log;

import java.util.List;

import in.smslite.SMSApplication;
import in.smslite.activity.CompleteSmsActivity;
import in.smslite.contacts.Contact;
import in.smslite.contacts.PhoneContact;
import in.smslite.db.Message;
import in.smslite.db.MessageDatabase;

/**
 * Created by rahul1993 on 2/4/2018.
 */

public class ThreadUtils {
  private static final String TAG = ThreadUtils.class.getSimpleName();
  public static class getUnreadSmsAysnTask extends AsyncTask<Void, Void, Integer> {
    Contact contact;
    Context context;

    getUnreadSmsAysnTask(Contact contact, Context context) {
      this.contact = contact;
      this.context = context;
    }

    @Override
    protected Integer doInBackground(Void... voids) {
      MessageDatabase mDB = MessageDatabase.getInMemoryDatabase(context);
      Cursor cursor = mDB.messageDao().getUnseenSmsCount(contact.getCategory());
      int countUnreadSMS = cursor.getCount();
      return countUnreadSMS;
    }
  }

  public static class getNotiSummaryAsyncTask extends AsyncTask<Void, Void, List<Message>> {
    Context context;
    int category;

    getNotiSummaryAsyncTask(int category, Context context) {
      this.category = category;
      this.context = context;
    }

    @Override
    protected List<Message> doInBackground(Void... voids) {
      MessageDatabase mDB = MessageDatabase.getInMemoryDatabase(context);
      List<Message> notificationSummary = mDB.messageDao().getNotificationSummary(category);
      return notificationSummary;
    }
  }

  public static class UpdateDbNotiClickedThread extends Thread {
    String address;
    public UpdateDbNotiClickedThread(String address){
      this.address =  address;
    }
    @Override
    public void run() {
      super.run();
      CompleteSmsActivity.completeSmsActivityViewModel.markAllRead(address);
    }
  }

  public static class UpdateMessageCategory extends Thread{
    Context context;
    List<Message> selectedItem;
    int category;
    int presentCategory;
    boolean checked;
    public UpdateMessageCategory(Context context, List<Message> selectedItem, int category, int presentCategory, boolean checked){
      this.context = context;
      this.selectedItem = selectedItem;
      this.category = category;
      this.checked = checked;
      this.presentCategory = presentCategory;
    }
    @Override
    public void run() {
      super.run();
      int length = selectedItem.size();
      MessageDatabase mDB = MessageDatabase.getInMemoryDatabase(context);
      for (int i = 0; i < length; i++) {
        Log.d(TAG, selectedItem.get(i).getAddress());
        mDB.messageDao().moveToCategory(selectedItem.get(i).getAddress(), category, presentCategory);
        if(checked){
          mDB.messageDao().updateSendFutureMessage(selectedItem.get(i).getAddress(), 1);
          mDB.messageDao().updateFutureCategory(selectedItem.get(i).getAddress(), category);
        } else {
//          mDB.messageDao().updateSendFutureMessage(selectedItem.get(i).getAddress(), 0);
        }
      }
    }
  }

  public static class MarkAllReadThread extends Thread{
    MessageDatabase mDB = MessageDatabase.getInMemoryDatabase(SMSApplication.getApplication());
    public MarkAllReadThread(){

    }
    @Override
    public void run() {
      super.run();
//      mDB.messageDao().markAllRead();
      ContentValues contentValues = new ContentValues();
      contentValues.put(Telephony.TextBasedSmsColumns.READ, 1);
      contentValues.put(Telephony.TextBasedSmsColumns.SEEN, 1);
      int updatedRows = SMSApplication.getApplication().getContentResolver().update(Telephony.Sms.CONTENT_URI, contentValues, null, null);
      Log.d(TAG, Integer.toString(updatedRows) + " updated rows");
    }
  }

  public static class cachePrimaryContactName extends Thread{

    public cachePrimaryContactName() {

    }

    @Override
    public void run() {
      super.run();
      PhoneContact.init(SMSApplication.getApplication());
      List<Message> list = MessageDatabase.getInMemoryDatabase(SMSApplication.getApplication())
          .messageDao().getPrimaryMessage();
      int size = list.size();
      for(int i = 0; i<size; i++){
        Contact contact = PhoneContact.get(list.get(i).getAddress(),true);
      }
    }
  }

  /*public static class sendFutureMessage extends Thread{
    String address;
    public sendFutureMessage(String address) {
      this.address = address;
    }

    @Override
    public void run() {
      super.run();
      MessageDatabase.getInMemoryDatabase(SMSApplication.getApplication()).messageDao().sendFutureMessage(address);
    }
  }*/
}

File: app/src/main/java/in/smslite/utils/ThreadUtils.kt
package `in`.smslite.utils

/**
 * Created by rahul1993 on 2/4/2018.
 */
//object ThreadUtils {
//    private val TAG = ThreadUtils::class.java.simpleName
//
//    class getUnreadSmsAysnTask internal constructor(var contact: Contact, var context: Context) : AsyncTask<Void?, Void?, Int>() {
//        protected override fun doInBackground(vararg void: Void?): Int {
//            val mDB: MessageDatabase = MessageDatabase.getInMemoryDatabase(context)
//            val cursor: Cursor? = mDB.messageDao().getUnseenSmsCount(contact.category)
//            return cursor!!.count
//        }
//    }
//
//    class getNotiSummaryAsyncTask internal constructor(var category: Int, var context: Context) : AsyncTask<Void?, Void?, List<Message>>() {
//        protected override fun doInBackground(vararg voids: Void?): List<Message> {
//            val mDB: MessageDatabase = MessageDatabase.getInMemoryDatabase(context)
//            return mDB.messageDao().getNotificationSummary(category)
//        }
//    }
//
//    class UpdateDbNotiClickedThread(var address: String) : Thread() {
//        override fun run() {
//            super.run()
//            CompleteSmsActivity.completeSmsViewModel?.markAllRead(address)
//        }
//    }
//
//    class UpdateMessageCategory(var context: Context, var selectedItem: List<Message>, var category: Int, var presentCategory: Int, var checked: Boolean) : Thread() {
//        override fun run() {
//            super.run()
//            val length = selectedItem.size
//            val mDB: MessageDatabase = MessageDatabase.getInMemoryDatabase(context)
//            for (i in 0 until length) {
//                Log.d(TAG, selectedItem[i].getAddress())
//                mDB.messageDao().moveToCategory(selectedItem[i].getAddress(), category, presentCategory)
//                if (checked) {
//                    mDB.messageDao().updateSendFutureMessage(selectedItem[i].getAddress(), 1)
//                    mDB.messageDao().updateFutureCategory(selectedItem[i].getAddress(), category)
//                } else {
////          mDB.messageDao().updateSendFutureMessage(selectedItem.get(i).getAddress(), 0);
//                }
//            }
//        }
//    }
//
//    class MarkAllReadThread : Thread() {
//        var mDB: MessageDatabase = MessageDatabase.getInMemoryDatabase(SMSApplication.application)
//        override fun run() {
//            super.run()
//            //      mDB.messageDao().markAllRead();
//            val contentValues = ContentValues()
//            contentValues.put(Telephony.TextBasedSmsColumns.READ, 1)
//            contentValues.put(Telephony.TextBasedSmsColumns.SEEN, 1)
//            val updatedRows: Int = SMSApplication.application?.getContentResolver()?.update(Telephony.Sms.CONTENT_URI, contentValues, null, null)
//                    ?: 1
//            Log.d(TAG, Integer.toString(updatedRows) + " updated rows")
//        }
//    }
//
//    class cachePrimaryContactName : Thread() {
//        override fun run() {
//            super.run()
//            PhoneContact.initialize(SMSApplication.application?.applicationContext!!)
//            val list: List<Message> = MessageDatabase.getInMemoryDatabase(SMSApplication.application)
//                    .messageDao().primaryMessage
//            val size = list.size
//            for (i in 0 until size) {
//                val contact: Contact = PhoneContact.get(list[i].getAddress(), true)
//            }
//        }
//    }
//}